Unrecognized /p: argument with nightly sdk

Using the nightly sdk, I can't pass a custom property with `dotnet publish`:

```
dotnet publish Benchmarks.csproj -c Release -o /tmp/benchmarks-agent/benchmarks-server-1/uu1yclhk.4ia/benchmarks/src/Benchmarks/published /p:MicrosoftNETCoreAppPackageVersion=5.0.0-rtm.20509.6 /p:MicrosoftAspNetCoreAppPackageVersion=5.0.0-rtm.20512.8 /p:BenchmarksNETStandardImplicitPackageVersion=5.0.0-rtm.20512.8 /p:BenchmarksNETCoreAppImplicitPackageVersion=5.0.0-rtm.20512.8 /p:BenchmarksRuntimeFrameworkVersion=5.0.0-rtm.20509.6 /p:BenchmarksTargetFramework=net5.0 /p:BenchmarksAspNetCoreVersion=5.0.0-rtm.20512.8 /p:MicrosoftAspNetCoreAllPackageVersion=5.0.0-rtm.20512.8 /p:NETCoreAppMaximumVersion=99.9 /p:MicrosoftNETCoreApp50PackageVersion=5.0.0-rtm.20509.6 /p:GenerateErrorForMissingTargetingPacks=false /p:MicrosoftNETPlatformLibrary=Microsoft.NETCore.App /p:RestoreNoCache=true --framework net5.0 --self-contained -r linux-x64 
Microsoft (R) Build Engine version 16.9.0-preview-20561-01+56ab63409 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1008: Only one project can be specified.
Switch: MicrosoftAspNetCoreAppPackageVersion=5.0.0-rtm.20512.8

For switch syntax, type "MSBuild -help"
Exit code: 1
```

--- a/src/Cli/dotnet/CommonOptions.cs
+++ b/src/Cli/dotnet/CommonOptions.cs
@@ -13,9 +13,9 @@ namespace Microsoft.DotNet.Cli
     internal static class CommonOptions
     {
         public static Option PropertiesOption() =>
-            new Option<string>(new string[] { "-property", "/p" })
+            new Option<string[]>(new string[] { "-property", "/p" })
             {
-                IsHidden = true
+                IsHidden = true,
             }.ForwardAsProperty();
 
         public static Option VerbosityOption() =>

--- a/src/Cli/dotnet/OptionForwardingExtensions.cs
+++ b/src/Cli/dotnet/OptionForwardingExtensions.cs
@@ -17,8 +17,8 @@ namespace Microsoft.DotNet.Cli
 
         public static Option ForwardAsSingle<T>(this Option<T> option, Func<T, string> format) => new ForwardedOption<T>(option).SetForwardingFunction(format);
 
-        public static Option ForwardAsProperty(this Option<string> option) => new ForwardedOption<string>(option)
-            .SetForwardingFunction((o) => new string[] { $"{option.Aliases.FirstOrDefault()}:{o.Replace("roperty:", string.Empty)}" });
+        public static Option ForwardAsProperty(this Option<string[]> option) => new ForwardedOption<string[]>(option)
+            .SetForwardingFunction((optionVals) => optionVals.SelectMany(optionVal => new string[] { $"{option.Aliases.FirstOrDefault()}:{optionVal.Replace("roperty:", string.Empty)}" }));
 
         public static Option ForwardAsMany<T>(this Option<T> option, Func<T, IEnumerable<string>> format) => new ForwardedOption<T>(option).SetForwardingFunction(format);
 

--- a/src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetPublishInvocation.cs
+++ b/src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetPublishInvocation.cs
@@ -94,5 +94,17 @@ namespace Microsoft.DotNet.Cli.MSBuild.Tests
                    .Should()
                    .Be($"{ExpectedPrefix} -target:Publish -property:NoBuild=true");
         }
+
+        [Fact]
+        public void CommandAcceptsMultipleCustomProperties()
+        {
+            var msbuildPath = "<msbuildpath>";
+            var command = PublishCommand.FromArgs(new[] { "/p:Prop1=prop1", "/p:Prop2=prop2" }, msbuildPath);
+
+            command.GetProcessStartInfo()
+               .Arguments
+               .Should()
+               .Be($"{ExpectedPrefix} -restore -target:Publish -property:Prop1=prop1 -property:Prop2=prop2");
+        }
     }
 }

